modules/agent/pkg/controllers/bundledeployment/controller.go

```diff
@@ -252,3 +277,113 @@ func readyError(status fleet.BundleDeploymentStatus) error {
 
 	return errors.New(msg)
 }
+
+func hasAutoReapply(bd *fleet.BundleDeployment, status *fleet.BundleDeploymentStatus) bool {
+	if bd.Spec.Options.Resync || bd.Spec.StagedOptions.Resync {
+		if bd.Spec.Options.ResyncPolicy == nil && bd.Spec.StagedOptions.ResyncPolicy == nil {
+			if status.LastApply == nil {
+				return false
```

Comment:
This doesn't look correct.... if lastApply is nil and resync is true then we return false? Look like it will never be deployed since that's the initail state..

Reply:
It will be deployed.

Right now there are two handles for BundleDeployments, DeployBundle and MonitorBundle.

The DeployBundle will deploy the bundle the first time and set the lastApply timestamp. I just wanted MonitorBundle to ignore reapply / re-queueing the bundle until the first deploy has happened.